### ISSUE TITLE
Data loader code

### ISSUE THREAD

**Issue by shubhamagarwal92** [woozzu/tagan#11]
Hi @woozzu, 

- Could you explain how are you creating the negative text [here](https://github.com/woozzu/tagan/blob/master/train.py#L124)? 

```

            img = img.mul(2).sub(1)
            # BTC to TBC
            txt = txt.transpose(1, 0)
            # negative text
            txt_m = torch.cat((txt[:, -1, :].unsqueeze(1), txt[:, :-1, :]), 1)
            len_txt_m = torch.cat((len_txt[-1].unsqueeze(0), len_txt[:-1]))
```

AFAIK, you are taking the last element and putting it at the top and then shifting others down. So, is this kind of randomization that you wanted to put? Any specific reason you didn't just use shuffle like:
 
```
txt.index_select(0, torch.randperm(2))
```

- Related to #10, Is it the same reason that you are doing it [here](https://github.com/woozzu/tagan/blob/master/train.py#L120): 
```
img = img.mul(2).sub(1)
```
It seems to me like an affine transformation to the image, which might be the reason that the images are rotated when visualized?  

- If I want to use my own dataset, do I have to do this kind of normalization for images as well? Basically, I want to use RGB Images of size 224X224. Could you tell what modifications do I have to make?

**Comment by woozzu** [comment]
@shubhamagarwal92 You can randomly shuffle the tensor. I wrote the code when pytorch does not support a random indexing (maybe v0.2?).

Yes, it is also for normalization. All you need to do for your own dataset is to normalize images to [-1, 1) and apply [this transform](https://github.com/woozzu/tagan/blob/master/train.py#L81).
